lib/datasource/git-refs/index.ts

```diff
@@ -10,6 +10,11 @@ const cacheMinutes = 10;
 // git will prompt for known hosts or passwords, unless we activate BatchMode
 process.env.GIT_SSH_COMMAND = 'ssh -o BatchMode=yes';
 
+const refMatchPattern = /(?<type>\w+)\/(?<value>.*)/;
+const lsRemoteReplace = /^.+?refs\//gm;
```

Comment:
Sorry to say, global matching regex needs to be reinstantiated, otherwise it will not work on next run
```suggestion
```

Reply:
I have reverted the previous commit and just extracted the ref-match out of the loop, which has the most performance impact.
The other regex patterns should have neglible performance impact especially compared to the io-delay of making remote (git) calls.